### ISSUE TITLE
object_detection trainer.py does not use train_config.load_all_detection_checkpoint_vars

### ISSUE THREAD

**Issue by DominikAuras** [tensorflow/models#3412]
There might a be bug `trainer.py`. In

https://github.com/tensorflow/models/blob/af6527c975a65a834d71adfde1041333b6fb8dc2/research/object_detection/trainer.py#L270-L271

the parameter `load_all_detection_checkpoint_vars`, which is defined here

https://github.com/tensorflow/models/blob/af6527c975a65a834d71adfde1041333b6fb8dc2/research/object_detection/protos/train.proto#L41

is not honored.

What I did: finetune mobilenet_ssd based on weights from your website. Then, use the last checkpoint to experiment with quantization.  So, I created a new train config, then referenced the checkpoint as "fine_tune_checkpoint". The initial loss is quite high, without quantization enabled! It appears that all the BoxPredictor scopes are not restored. Indeed, after checking the code, this is expected. Only the FeatureExtractor scope gets restored. The additional flag mentioned above needs to be used, I believe. When I changed the aforementioned code line in trainer.py, it started working as expecting. From Step 1 on in the new training (in a different train dir btw.), training continued exactly from the last training state.

**Comment by asimshankar** [comment]
@jch1 @tombstone @derekjchow @jesu9 @dreamdragon : Mind taking a look?

**Comment by rogercooper76** [comment]
Hi @DominikAuras , Thanks for your sharing. I think I bumped into the similar issue. Would you mind elaborating a little bit more how you fix this? Did you just change the default value of load_all_detection_checkpoint_vars from false to true in .proto file? Or, is there any more elegant way to achieve it? Thank you in advance.

**Comment by DominikAuras** [comment]
@rogercooper76 
I have changed the lines

```python
      var_map = detection_model.restore_map(
          from_detection_checkpoint=train_config.from_detection_checkpoint)
```

to

```python
      var_map = detection_model.restore_map(
          from_detection_checkpoint=train_config.from_detection_checkpoint,
          load_all_detection_checkpoint_vars=train_config.load_all_detection_checkpoint_vars)
```

This also restores `global_step`. If that is not what you want, you might change the next lines too:

```python
      available_var_map = (variables_helper.
                           get_variables_available_in_checkpoint(
                               var_map, train_config.fine_tune_checkpoint, include_global_step=False))
```

This is a workaround.

**Comment by rogercooper76** [comment]
Thanks a lot, @DominikAuras. It works! Those are all quite useful information.

**Comment by rogercooper76** [comment]
Hi @DominikAuras, sorry for bugging you. After I successfully loaded all parameters, the detection precision was dropping throughout the training with exactly same network (I also posted an issue [here](https://github.com/tensorflow/models/issues/3384)). May I know whether you observe similar issue or not? Thanks.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
